ironfist/xml/map_xml.xsd

```diff
@@ -80,17 +79,44 @@
       <xs:element name="field_E8" type="xs:int"/>
     </xs:sequence>
   </xs:complexType>
-  
+
+  <xs:complexType name="tableElement_t">
+    <xs:attribute name="key" type="xs:string"/>
+    <xs:attribute name="value" type="xs:string"/>
+    <xs:attribute name="type" type="xs:string" default="string"/>
+  </xs:complexType>
+
+  <xs:complexType name="table_t">
+    <xs:sequence>
+      <xs:element name="tableId" type="xs:ID"/>
+      <xs:element name="tableElement" type="tableElement_t" minOccurs="0" maxOccurs="unbounded"/>
+      <xs:element name="parentTable" type="xs:IDREF" minOccurs="0" maxOccurs="1"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="mapVariable_t">
+    <xs:sequence>
+      <xs:element name="table" type="table_t" minOccurs="0" maxOccurs="unbounded"/>

```

Comment:
The issue here is not to break old saves. The attributes value and id have to stay the same. 
As far as recursiveness is concerned, the way I designed it a table has elements and subtables (defined by the IDs of those subtables). I guess it is not the purest of recursions, I'll try to change that.